### DISCUSSION TITLE
Serve a PDF file

### DISCUSSION THREAD

**Discussion by Martien** [observablehq/framework#1859] [Q&A] [answered]
I have some pages that include a link to a PDF file, yet the preview server apologises: Sorry, but we can’t find the page you requested.

How can I make the (preview) server also serve PDF files (or any type for that matter)?

**Comment by mbostock** [comment, marked as answer]
You should be able to use an `<a href="…" download>` link.

**Reply by Martien** [reply]
🙏 Works! 🎉 I feel a little bit stupid – I could have searched for it.

**Reply by mbostock** [reply]
Don’t feel bad; I don’t think it’s expressly documented. The closest I could find is under [Files > Routing](https://observablehq.com/framework/files#routing) where it says “Some additional assets are automatically promoted to file attachments…” but doesn’t specifically mention `<a href download>`. I believe the list of supported attributes is here:

https://github.com/observablehq/framework/blob/8efdcf45c00f7cb471deab01f65fe93e1b539d44/src/html.ts#L8-L18

Related #639 #1328.

**Comment by Martien** [comment]
BTW, I am falling in love with Framework. Have already imported 89 Wordpress files.

**Comment by Fil** [comment]
Also related https://github.com/observablehq/framework/issues/1694
